flink-staging/flink-gelly/pom.xml

```diff
@@ -52,4 +52,35 @@ under the License.
 			<scope>test</scope>
 		</dependency>
 	</dependencies>
+
+    <!-- See main pom.xml for explanation of profiles -->
+    <profiles>
+        <profile>
+            <id>hadoop-1</id>
+            <activation>
+                <property>
+                    <!-- Please do not remove the 'hadoop1' comment. See ./tools/generate_specific_pom.sh -->
+                    <!--hadoop1--><name>hadoop.profile</name><value>1</value>
+                </property>
+            </activation>
+            <dependencies>
+                <!-- Add this here, for hadoop-2 we don't need it since we get guava transitively -->
+                <dependency>
+                    <groupId>com.google.guava</groupId>
+                    <artifactId>guava</artifactId>
+                    <version>${guava.version}</version>
+                    <scope>provided</scope>
+                </dependency>
+            </dependencies>
+        </profile>
+        <profile>
```

Comment:
Regarding the pom, we've actually merged this change as part of #429. We added the guava dependency after [this discussion](https://mail-archives.apache.org/mod_mbox/flink-dev/201502.mbox/%3CCAJAsdNijOSZwqnn1ugw0qwJvXurmjb%3DOz64-sd3u97gZcbcW0Q%40mail.gmail.com%3E) in the dev mailing list. If there's a better way to deal with this, please let me know! The hadoop2 profile seems to be there by mistake, I'll fix that :)